### PR TITLE
Add-sign-up-endpoint

### PR DESCRIPTION
**Pull request by SOY4RIAS**
> ⚠️ first this PR ➡️ #3 

### Pull request type

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

---

### Description

- Added a implementation of sign up endpoint
- [Pending] add unit tests in a following PR

---

#### Board reference

- [board](link)

### CODE REVIEW

**Review by brunoBrizo — APPROVED** [review]
LGTM

**Review by matiasirpac — APPROVED** [review]
Looks good, just remember to use path alias

File: src/shared/decorators/match.decorator.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  ValidationOptions,
+  ValidatorConstraint,
+  ValidatorConstraintInterface,
+  equals,
+  registerDecorator,
+} from 'class-validator';
+
+@ValidatorConstraint({ name: 'Match' })
+export class MatchConstraint implements ValidatorConstraintInterface {
+  validate(value: any, validationArguments?: ValidationArguments): boolean {
```
Comment (matiasirpac) [inline src/shared/decorators/match.decorator.ts]: You could use "unknown". Let me know if I'm wrong.

Comment (SOY4RIAS) [reply src/shared/decorators/match.decorator.ts]: you are absolutely right 🚀

**Pull request merged**